### PR TITLE
azure vnet

### PR DESCRIPTION
**Pull request by felipesborges**
There are a lot to improvments to be done,  but i'm providing the files anyway.

### CODE REVIEW

File: azure-resource-group/main.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+provider "azurerm" {
+  features {}
+  }
+
+resource "azurerm_resource_group" "test" {
+  name     = var.resource_group_name
+  location = var.location
```
Comment (benhur-araujo) [inline azure-resource-group/main.tf:7]:
```suggestion
  location = var.location
  tags = var.tags
```

File: azure-resource-group/variables.tf
Comment (benhur-araujo) [inline azure-resource-group/variables.tf:1]: Need to add a tag variable

File: azure-vnet/main.tf
Comment (benhur-araujo) [inline azure-vnet/main.tf:1]: missing tag variable here as well

**Comment by benhur-araujo** [comment]
You'll need to create a terraform block for both modules. E.g:
- Create a new file named "versions.tf" on each module

terraform {
  required_version = "~> 1.0"
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 3.82"
    }
  }
}

**Comment by benhur-araujo** [comment]
Please use the Conventional Commits standard for commit messages:
- https://blog.rocketseat.com.br/como-fazer-um-commit-conventional-commits/
- https://www.conventionalcommits.org/en/v1.0.0/

For this PR, the suggestion would be to separate the PR into two separate commits, because you are creating two different modules:
- feat: new module for az resource-group
- feat: new module for az vnet

File: azure-vnet/main.tf [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  }
+
+resource "azurerm_virtual_network" "test" {
+  name                = var.vnet_name
+  address_space       = var.vnet_address_space
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  }
+
+resource "azurerm_subnet" "test" {
```
Comment (benhur-araujo) [inline azure-vnet/main.tf]: we'll need to create multiple subnets for this vnet (one for the k8s and another for the VMs, for example). How can we declare this resource in a way that allows us to create multiple subnets only by editing the terragrunt.hcl file?

**Comment by felipesborges** [comment]
> terraform {
> required_version = "~> 1.0" required_providers { azurerm = { source = "hashicorp/azurerm" version = "~> 3.82"

i saw the versions.tf but had no idea (still haven't tbh) how to determine the minimal version is necessary for each scenario... will study that later

**Comment by felipesborges** [comment]
> For this PR, the suggestion would be to separate the PR into two separate commits, because you are creating two different modules:

sorry man, not used to do PRs, long way to go.... thanks for the tips

File: azure-vnet/main.tf [hunk 3]
```diff
@@ -2,16 +2,20 @@ provider "azurerm" {
...
+resource "azurerm_subnet" "subnets" {
+  for_each             = var.subnets
   resource_group_name  = var.resource_group_name
   virtual_network_name = var.vnet_name
-  address_prefixes     = var.subnet_prefixes
-  }
\ No newline at end of file
+  name                 = each.value.name
+  address_prefixes     = each.value.address_prefix
+#  tags                 = var.tags - Can't have tags on the subnets
```
Comment (benhur-araujo) [inline azure-vnet/main.tf]:
```suggestion

```

**Comment by leohauschild** [comment]
> > terraform {
> > required_version = "~> 1.0" required_providers { azurerm = { source = "hashicorp/azurerm" version = "~> 3.82"
> 
> i saw the versions.tf but had no idea (still haven't tbh) how to determine the minimal version is necessary for each scenario... will study that later

I think we can use this approach https://developer.hashicorp.com/terraform/language/expressions/version-constraints#1-1

**Review by leohauschild — APPROVED** [review]

**Pull request merged**